migrator/src/main/java/com/redhat/lightblue/migrator/AbstractController.java

```diff
@@ -27,16 +27,17 @@
     protected final Locking locking;
     protected final Class migratorClass;
     protected final ThreadGroup migratorThreads;
-    protected final HashSet<String> myLocks = new HashSet<>();
-
-    public AbstractController(Controller controller, MigrationConfiguration migrationConfiguration, String threadGroupName) {
-        this.migrationConfiguration = migrationConfiguration;
-        this.controller = controller;
-        lbClient = controller.getLightblueClient();
-        locking = lbClient.getLocking("migration");
-        if (migrationConfiguration.getMigratorClass() == null) {
-            migratorClass = DefaultMigrator.class;
-        } else {
+    protected final HashSet<String> myLocks=new HashSet<>();
+    protected boolean stopped=false;
```

Comment:
It is not read here, but when the thread needs to stop, it is set at this level. All controller classes inherit AbstractController.